### PR TITLE
xapi-guard cleanups

### PR DESCRIPTION
**Pull request by psafont**
These are the less controversial parts of https://github.com/xapi-project/xen-api/pull/4985

We can merge these and the blobstorage can be reviewed at a later time.

I'm interested in merging these since I'm also working in the xapi-guard area and I'd rather rebase on top of these that the other way round once edwin is back

### CODE REVIEW

**Comment by psafont** [comment]
I've separated httpsvr into to private libraries so the server implementation is not included in most places where the client is needed, and made xapi-database internal as well to be able to make the other two internals as well.

**Comment by robhoes** [comment]
SpecSavers would have opened up the opportunity to make more libraries internal to this repo.

**Comment by psafont** [comment]
Agh, can't make all the libraries private, they're being used by a separate daemon =_=

File: README.markdown [hunk 1]
```diff
@@ -58,9 +58,9 @@ To build xen-api from source, we recommend using [opam](https://opam.ocaml.org/d
 6) Install all the Packages.
 
     ```bash
-    PACKAGES="xapi-cli-protocol xapi-client xapi-consts xapi-database xapi-datamodel xapi-types xapi xe xen-api-sdk xen-api-client xen-api-client-lwt xen-api-client-async xapi-rrdd xapi-rrdd-plugin xapi-rrd-transport xapi-rrd-transport-utils rrd-transport rrdd-plugin rrdd-plugins rrddump gzip http-svr pciutil safe-resources sexpr stunnel uuid xapi-compression xml-light2 zstd vhd-tool xs-toolstack"
+    PACKAGES="xapi-cli-protocol xapi-client xapi-consts xapi-datamodel xapi-types xapi xe xen-api-sdk xen-api-client xen-api-client-lwt xen-api-client-async xapi-rrdd xapi-rrdd-plugin xapi-rrd-transport xapi-rrd-transport-utils rrd-transport rrdd-plugin rrdd-plugins rrddump gzip http-lib pciutil safe-resources sexpr stunnel uuid xapi-compression xml-light2 zstd vhd-tool xs-toolstack"
 
-    # NOT needed with opam>=2.1.0) Install all the dependances (Including OS):
+    # NOT needed with opam>=2.1.0) Install all the dependences (Including OS):
```
Comment (lindig) [inline README.markdown:63]: This is still wrong: it's `dependencies`

**Review by lindig — APPROVED** [review]

**Pull request merged**